### PR TITLE
Rename beacon-chain/core/state to beacon-chain/core/transition pkg

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

General code clean up. We have been aliasing this package to "core" since there are other "state" packages. 

**Which issues(s) does this PR fix?**

None filed.

**Other notes for review**

Fixes `core2` imports from #9519

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Goland renamed a bunch of comments... fixing before requesting review again

File: beacon-chain/sync/initial-sync/blocks_fetcher_peers_test.go [hunk 1]
```diff
@@ -286,8 +286,8 @@ func TestBlocksFetcher_filterPeers(t *testing.T) {
 			// such peers can be selected. That's why they are sorted here.
 			sort.SliceStable(filteredPIDs, func(i, j int) bool {
 				score1 := fetcher.p2p.Peers().Scorers().BlockProviderScorer().Score(filteredPIDs[i])
-				score2 := fetcher.p2p.Peers().Scorers().BlockProviderScorer().Score(filteredPIDs[j])
-				if score1 == score2 {
+				score := fetcher.p2p.Peers().Scorers().BlockProviderScorer().Score(filteredPIDs[j])
+				if score1 == score {
```
Comment (rkapka) [inline beacon-chain/sync/initial-sync/blocks_fetcher_peers_test.go:290]: this change should be reverted

Comment (rkapka) [inline beacon-chain/sync/initial-sync/blocks_fetcher_peers_test.go:290]: This change should be reverted.

File: beacon-chain/sync/initial-sync/round_robin_test.go [hunk 2]
```diff
@@ -567,10 +567,10 @@ func TestService_blockProviderScoring(t *testing.T) {
 	scorer.IncrementProcessedBlocks(peer2, 1)
 	scorer.IncrementProcessedBlocks(peer3, 1)
 	score1 := scorer.Score(peer1)
-	score2 := scorer.Score(peer2)
+	score := scorer.Score(peer2)
```
Comment (rkapka) [inline beacon-chain/sync/initial-sync/round_robin_test.go:570]: This change should be reverted.

File: shared/aggregation/attestations/attestations_test.go [hunk 3]
```diff
@@ -388,11 +388,11 @@ func TestAggregateAttestations_PerformanceComparison(t *testing.T) {
 
 			atts, err = MaxCoverAttestationAggregation(generateAtts(tt.bitsList))
 			require.NoError(t, err)
-			score2 := scoreAtts(atts)
+			score := scoreAtts(atts)
```
Comment (rkapka) [inline shared/aggregation/attestations/attestations_test.go:391]: This change should be reverted.

**Review by rkapka — CHANGES_REQUESTED** [review]
Found a few unwanted changes.

**Review by rkapka — APPROVED** [review]

**Pull request merged**